x-pack/plugins/security_solution/public/onboarding/components/onboarding_body/cards/integrations/available_packages.tsx

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useCallback, useState } from 'react';
+
+import useAsyncRetry from 'react-use/lib/useAsyncRetry';
+import type { AvailablePackagesHookType } from '@kbn/fleet-plugin/public';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { i18n } from '@kbn/i18n';
+import { EuiButton, EuiCallOut, EuiSkeletonText } from '@elastic/eui';
+import { fetchAvailablePackagesHook } from './utils';
+import { PackageListGrid } from './package_list_grid';
+import { LOADING_SKELETON_HEIGHT } from './const';
+
+export const AvailablePackages = React.memo(() => {
+  const [fetchAvailablePackages, setFetchAvailablePackages] = useState<AvailablePackagesHookType>();
+
+  const { error, retry, loading } = useAsyncRetry(async () => {
+    if (fetchAvailablePackages) {
+      return;
+    }
+    const loadedHook = await fetchAvailablePackagesHook();
+    setFetchAvailablePackages(() => {
+      return loadedHook;
+    });
+  });
+
+  const onRetry = useCallback(() => {
+    if (!loading) {
+      retry();
+    }
+  }, [loading, retry]);
+
+  if (error) {
+    return (
+      <EuiCallOut
+        title={i18n.translate('xpack.securitySolution.onboarding.asyncLoadFailureCallout.title', {
+          defaultMessage: 'Loading failure',
+        })}
+        color="warning"
+        iconType="cross"
+        size="m"
+      >
+        <p>
+          <FormattedMessage
+            id="xpack.securitySolution.onboarding.asyncLoadFailureCallout.copy"
+            defaultMessage="Some required elements failed to load."
+          />
+        </p>
+        <EuiButton color="warning" data-test-subj="retryButton" onClick={onRetry}>
+          <FormattedMessage
+            id="xpack.securitySolution.onboarding.asyncLoadFailureCallout.buttonContent"
+            defaultMessage="Retry"
+          />
+        </EuiButton>
+      </EuiCallOut>
+    );
+  }
+  if (loading || !fetchAvailablePackages) {
+    return (
+      <EuiSkeletonText
+        data-test-subj="loadingPackages"
+        isLoading={true}
+        lines={LOADING_SKELETON_HEIGHT}
+      />
+    );
+  }
+  return <PackageListGrid useAvailablePackages={fetchAvailablePackages} />;
```

Comment:
This requires a lot of custom logic to load a hook from Fleet. What if we create a HOC to do that, so we can clean this component from that hook lazy loading logic, this way we can also reuse it if we ever need to do this somewhere else.
We can create a new file at `x-pack/plugins/security_solution/public/common/components/with_lazy_hook/index.tsx` with:
```
export const withLazyHook = <D extends {}, P extends {}>(
  Component: React.ComponentType<PropsWithChildren<D & P>>,
  moduleImport: () => Promise<P>,
  fallback?: React.ReactNode
) => {
  return React.memo(function WithLazy(props: D) {
    const [lazyModuleProp, setLazyModuleProp] = useState<P>();

    useEffect(() => {
      moduleImport().then((module) => {
        setLazyModuleProp(() => module);
      });
    }, []);

    return lazyModuleProp ? <Component {...lazyModuleProp} {...props} /> : <>{fallback}</> ?? null;
  });
};
 ```
 
Then we could just do:
```
export const AvailablePackages = withLazyHook(
  PackageListGrid,
  () => import('@kbn/fleet-plugin/public').then((module) => module.AvailablePackagesHook()),
  <EuiSkeletonText
    data-test-subj="loadingPackages"
    isLoading={true}
    lines={LOADING_SKELETON_HEIGHT}
  />
);
```
But I don't think we need to create a file for that, we can just place it next to the component definition and export the wrapped component.
We don't actually need error catching for lazy loading, we never do that, we assume all our libraries exist if the bundle compiles.